Jenkinsfile

```diff
@@ -588,6 +588,14 @@ pipeline {
                         buildHipClangJobAndReboot(setup_flags: MLIR_flags, build_env: extra_log_env, test_flags: ' --verbose ', gpu_arch: "gfx908", mlir_build: "ON")
                     }
                 }
+                stage('Fp32 OpenCL MLIR gfx908') {
+                    when { expression { params.TARGET_GFX908 } }
```

Comment:
```suggestion
                    when {
                        beforeAgent true
                        expression { params.TARGET_GFX908 && params.DATATYPE_FP32 }
                    }
```

Reply:
Commited as-is. I'm not very familiar with the beforeAgent clause. Doing a bit of the read up it seems to make sure the expression get evaluated before node is selected.  However, the node is also filtered against `gfx908` label, so why does the evaluation sequence matter?

Reply:
@jerryyin without it in Jenkins parallel stages it will wait till a node (Agent) is available and then evaluate whether it fits in the expression, causing delays and long wait times when the pipelines are congested.
CC: @atamazov

Reply:
Make sense, thanks